### ISSUE TITLE
Enhance a11y for `Table`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#1242]
path: stories/components/data-display/table.stories.tsx

### Description
1 accessibility violation was found in the `Table`.

#### [withDefaultSelectedRowIds](https://yamada-ui.github.io/yamada-ui/?path=/story/components-data-display-table--with-default-selected-row-ids)

Expected the HTML found at $('tr[data-selected=""]') to have no violations:
    <tr class="ui-table__tr css-1j26qlj" data-selected="" aria-checked="true">

    Received:
    Elements must only use supported ARIA attributes (aria-allowed-attr)

    Fix all of the following:
      ARIA attribute is not allowed: aria-checked="true"

    You can find more information on this issue here: 
    https://dequeuniversity.com/rules/axe/4.9/aria-allowed-attr?application=axeAPI

### Problem Statement/Justification
A11y is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all jobs, if there are too many jobs to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

**Comment by daichi000** [comment]
@hirotomoyamada Hi!  can i work on this issue?

**Comment by hirotomoyamada** [comment]
@daichi000 

Hi, thanks for the comment!

Yes, please.
